src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -249,34 +260,91 @@ describe('Transactions Controller (Unit)', () => {
     it('Should get a Custom transaction mapped to the expected format', async () => {
       const chainId = faker.random.numeric();
       const safeAddress = faker.finance.ethereumAddress();
+      const toAddress = faker.finance.ethereumAddress();
       const chainResponse = chainBuilder().with('chainId', chainId).build();
+      const safeAppUrl = faker.internet.url();
+      const safeAppLogoUri = faker.internet.url();
+      const safeAppName = faker.random.words();
+      const methodName = 'multiSend';
+      const contractDisplayName = faker.random.words();
+      const contractLogoUri = faker.internet.url();
+      const executionDate = faker.date.recent();
+      const safeTxHash = faker.datatype.hexadecimal(32);
+      const value = '0';
+      const nonce = faker.datatype.number({ min: 1 });
+      const confirmations = [
+        confirmationBuilder().build(),
+        confirmationBuilder().build(),
+        confirmationBuilder().build(),
+      ];
+      const safeAppsResponse = [
+        safeAppBuilder()
+          .with('url', safeAppUrl)
+          .with('iconUrl', safeAppLogoUri)
+          .with('name', safeAppName)
+          .build(),
+      ];
```

Comment:
I've removed some intermediate variables in [f714450](https://github.com/5afe/safe-client-gateway-nest/pull/236/commits/f71445053424d8dfd6d54d73e5b21c2517fbc5a5), but I'm not sure I can use the `MultisigTransaction` built directly on the response assertion, since the builder creates a domain object and the response is a route object 🤔 Maybe I'm missing something.

Reply:
Not sure if I understand. The `domainTransaction` in https://github.com/5afe/safe-client-gateway-nest/pull/236/files#diff-0d3b6a85b1cf9923c19fdfdc2297d006077468cf7450dea8d7cb8e4eacf572a6R278-R307

it's being used by just setting fake values. Why not use `multisigTransactionBuilder().build()`?

Reply:
Hmm, I think that some parts of `domainTransaction` are as you said just random data, but other are important to have them set to concrete values, since we need to assert in the response those values are mapped correctly. I mean, for example, we need to set a concrete `dataDecoded` value to assert `actionCount` in the response, we need to set both concrete `isExecuted` and `isSuccessful` to assert `txStatus` contains the expected value, etc. 

I think I've set the minimum amount of non-random values for `domainTransaction` in . Please let me know if you see this could be handled in a different way 🙂

Also, going through this, I think `MultisigTransactionBuilder` could be improved a little, because for example:
```javascript
.with('origin', faker.internet.url())
```
is not correct: `origin` should be a string containing a JSON. 

And also:
```javascript
.with('confirmations', [confirmationBuilder().build()])
``` 
could generate a random number (between 0 and 5 for example) of confirmations instead of just 1, this will make tests stronger. What do you think? I can create a new PR with these minor improvements 🙂

Reply:
> Hmm, I think that some parts of `domainTransaction` are as you said just random data, but other are important to have them set to concrete values, since we need to assert in the response those values are mapped correctly. I mean, for example, we need to set a concrete `dataDecoded` value to assert `actionCount` in the response, we need to set both concrete `isExecuted` and `isSuccessful` to assert `txStatus` contains the expected value, etc.
> 
> I think I've set the minimum amount of non-random values for `domainTransaction` in [a9d7769](https://github.com/5afe/safe-client-gateway-nest/pull/236/commits/a9d776911ee96604a21921ca6f63720492e3dee2). Please let me know if you see this could be handled in a different way 🙂

I do think that now it's better because it seems that the data that it's being generated is more targeted to what is required for the test 👍

> 
> Also, going through this, I think `MultisigTransactionBuilder` could be improved a little, because for example:
> could generate a random number (between 0 and 5 for example) of confirmations instead of just 1, this will make tests stronger. What do you think? I can create a new PR with these minor improvements 🙂

For this I'd suggest to create a PR as you mentioned 👍